lightning/src/routing/router.rs

```diff
@@ -719,8 +719,9 @@ where L::Target: Logger {
 			node_info.features.supports_basic_mpp()
 		} else { false }
 	} else { false };
-	log_trace!(logger, "Searching for a route from payer {} to payee {} {} MPP", our_node_pubkey,
-		payment_params.payee_pubkey, if allow_mpp { "with" } else { "without" });
+	log_trace!(logger, "Searching for a route from payer {} to payee {} {} MPP and {} first hops {}overriding the network graph", our_node_pubkey,
+		payment_params.payee_pubkey, if allow_mpp { "with" } else { "without" },
+		if first_hops.is_some() { first_hops.as_ref().unwrap().len() } else { 0 }, if first_hops.is_some() { "" } else { "not " });
```

Comment:
I think since we have the `0`, can't we get rid of `if first_hops.is_some() { "" } else { "not " }`?

Reply:
The behavior is different between `Some(0 length)` and `None`, though - `Some(0 length)` will always fail to route, `None` will look at the network graph.

Reply:
Hmm, I think we should error at the beginning of the method for that first case then?

Reply:
I mean we do error, indicating that we have no path to use. I guess we could update the error message as well, but its nice to log the info as well because users may or may not log specific error messages, relying on our own logs instead (which is kinda annoying, but seems to be a recurring problem).

Reply:
Ah, it does already get its own error message `Cannot route when there are no outbound routes away from us` is unique to this case.